webserver/CMakeLists.txt

```diff
@@ -1,22 +1,59 @@
 add_subdirectory(authenticators)
 
+add_compile_definitions(
+  MG_ENABLE_DEBUG=0
+
+  # Turn off some unneeded by default enabled features of Mongoose.
+  # Makes the binary smaller!
+  CS_DISABLE_MD5=1
+  CS_DISABLE_SHA1=1
+  MG_DISABLE_HTTP_DIGEST_AUTH=1
+  MG_ENABLE_ASYNC_RESOLVER=0
+  MG_ENABLE_CALLBACK_USERDATA=0
+  MG_ENABLE_DNS=0
+  MG_ENABLE_FILESYSTEM=1
+  MG_ENABLE_HTTP_WEBSOCKET=0
+  MG_ENABLE_HTTP_URL_REWRITES=0
+  MG_ENABLE_IPV6=0
+  MG_ENABLE_MQTT=0
+
+  # Only exposes a pthread_create() API, changes no behaviour w.r.t CodeCompass.
+  MG_ENABLE_THREADS=0
```

Comment:
I went purely by reading `mongoose.h` and `mongoose.c`. As the code is meant for embedded system, the header (and to some extent, the implementation code), is relatively straightforward to read.

Reply:
`MG_ENABLE_DEBUG` simply results in a very obscure dump of steps taken and enum values for each request. It's *somewhat* useful for debugging if you manually trace the requests and such, but not useful for production system. (It's set to 0 because is a **build-time** flag, due to how Mongoose's code is structured, you can't make this dependent on `--loglevel debug` at runtime.)

Mongoose is not just a server, it can also be used as a **client** library. `MG_ENABLE_DNS` (defaulting to `1`) toggles whether, if used *as a client*, the code has the ability to do DNS resolution. Because we are not using Mongoose as a client, we do not need this functionality.

(`_IPV6` and `_MQTT` similarly disable networking stacks that we are not using.)

`MG_ENABLE_ASYNC_RESOLVER` is related to this, too. (If enabled, *clientside* DNS resolution is done asynchronously, by default via `MG_DEFAULT_NAMESERVER "8.8.8.8"`.

`MG_ENABLE_FILESYSTEM` is to be used if we are serving a conventional HTTP root (i.e. like Apache `Options Index`) and static files. This can't be used, as we have special endpoints (for Thrift) which would conflict, if Mongoose was to serve the filesystem natively.

`MG_ENABLE_HTTP_URL_REWRITES` is explained pretty well in the header:
![Screenshot_20200728_154040](https://user-images.githubusercontent.com/1969470/88673298-ba9c0a80-d0e8-11ea-8857-ff4e817f62c7.png)

`MG_ENABLE_CALLBACK_USERDATA` is a flag that changes the signatures of Mongoose handlers to take an additional userdata argument, like this:
![Screenshot_20200728_154303](https://user-images.githubusercontent.com/1969470/88673560-0c449500-d0e9-11ea-94fb-3004e13d5985.png)
In many functions, if this is enabled, an additional `void *` argument is present in the functions' signatures, which allows registering/passing an arbitrary object through Mongoose.